move/axelar_gateway/sources/gateway.move

```diff
@@ -208,25 +204,26 @@ entry fun rotate_signers(
     proof_data: vector<u8>,
     ctx: &TxContext,
 ) {
+    let data = data_mut!(self);
     let weighted_signers = utils::peel!(
         new_signers_data,
         |bcs| weighted_signers::peel(bcs),
     );
     let proof = utils::peel!(proof_data, |bcs| proof::peel(bcs));
 
-    let enforce_rotation_delay = ctx.sender() != self.operator;
+    let enforce_rotation_delay = ctx.sender() != data.operator();
 
-    let is_latest_signers = self
-        .signers
+    let is_latest_signers = data
+        .signers()
         .validate_proof(
             data_hash(COMMAND_TYPE_ROTATE_SIGNERS, new_signers_data),
             proof,
         );
     assert!(!enforce_rotation_delay || is_latest_signers, ENotLatestSigners);
 
     // This will fail if signers are duplicated
-    self
-        .signers
+    data
```

Comment:
I don't like this pattern of keeping the data separate from the logic such that all data fields need to be exposed to be used in this logic, and especially exposing mutable references to internal data fields. How about we just define the logic that mutates `data` in the same module. For e.g. define a rotate_signers function in the data module, and pass in the typed `proof`, `weighted_signers`, `ctx`, and `clock`. UX will be cleaner. If we need to upgrade, then we can first just move the logic to the new V1 module, and then modify it as needed to work with the new object. It'll reduces the coupling between the public entrypoints and data specific logic. What do you think?